File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/java/bladi23.java
public class bladi23 {
    public static void main(String[] args) {

        // Tipos de operadores en java

        // Operadores aritmeticos
        int a = 2, b = 4;
        System.out.println("a + b = " + (a + b));
        System.out.println("a - b = " + (a - b));
        System.out.println("a * b = " + (a * b));
        System.out.println("b / a = " + (b / a));
        System.out.println("b % a = " + (b % a));
        System.out.println("a++ = " + (a++));
        System.out.println("b-- = " + (b--));
        // Operadores de asignacion
        a += 2;
        System.out.println("a += 2 = " + a);

        // Operadores de comparacion
        System.out.println("a == b: " + (a == b));
        System.out.println("a != b: " + (a != b));
        System.out.println("a > b: " + (a > b));
        System.out.println("a < b: " + (a < b));
        System.out.println("b >= a: " + (b >= a));
        System.out.println("b <= a: " + (b <= a));
        // Operadores logicos
        System.out.println("(a == b) && (a < b): " + ((a == b) && (a < b)));
        System.out.println("(a == b) || (a < b): " + ((a == b) || (a < b)));
        System.out.println("!(a == b): " + !(a == b));

        // Operadores de incremento y decremento
        a++; // incremento
        a--; // decremento

        // Operadores de concatenacion
        String texto1 = "Hola, ";
        String texto2 = "¿cómo estás?";
        System.out.println(texto1 + texto2);

        // Operadores ternarios
        String result = (a > b) ? "a es mayor que b" : "b es mayor o igual que a";
        System.out.println(result);
        // Operadores de instancia
        String texto = "Hola";
        if (texto instanceof String) {
            System.out.println("texto es una instancia de String");
        }

        // Tipos de estructura de control en java

        // Estructura if
        if (a > b) {
        } else if (a == b) {
        } else {
            System.out.println("a es menor que b");
        }

        // Estructura switch
        switch (a) {
            case 1:
                System.out.println("a es igual a 1");
                break;
            case 2:
                System.out.println("a es igual a 2");
                break;
            default:
                System.out.println("a no es igual a 1 ni a 2");
                break;
        }

        // Estructura while
        while (a < b) {
            System.out.println("a es menor que b");
            a++;
        }

        // Estructura do while
        do {
            System.out.println("a es menor que b");
            a++;
        } while (a < b);

        // Estructura for
        for (int i = 0; i < 10; i++) {
            System.out.println("i es igual a " + i);
        }

        // Estructura for each
        int[] numeros = { 1, 2, 3, 4, 5 };
        for (int numero : numeros) {
            System.out.println("numero es igual a " + numero);
        }

        // Estructura break
        for (int i = 0; i < 10; i++) {
            if (i == 5) {
                break;
            }
            System.out.println("i es igual a " + i);
        }
        // Estructura continue
        for (int i = 0; i < 10; i++) {
            if (i == 5) {
                continue;
            }
            System.out.println("i es igual a " + i);
        }

        /*
         * DIFICULTAD EXTRA (opcional):
         * Crea un programa que imprima por consola todos los números comprendidos
         * entre 10 y 55 (incluidos), pares, y que no son ni el 16 ni múltiplos de 3.
         */

        for (int i = 10; i <= 55; i++) {
            if (i % 2 == 0 && i != 16 && i % 3 != 0) {
                System.out.println(i);
            }
        }
    }
}

File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/python/bladi23.py
"""
 * EJERCICIO:
 * - Crea ejemplos utilizando todos los tipos de operadores de tu lenguaje:
 *   Aritméticos, lógicos, de comparación, asignación, identidad, pertenencia, bits...
 *   (Ten en cuenta que cada lenguaje puede poseer unos diferentes)
 * - Utilizando las operaciones con operadores que tú quieras, crea ejemplos
 *   que representen todos los tipos de estructuras de control que existan
 *   en tu lenguaje:
 *   Condicionales, iterativas, excepciones...
 * - Debes hacer print por consola del resultado de todos los ejemplos.
 *
 * DIFICULTAD EXTRA (opcional):
 * Crea un programa que imprima por consola todos los números comprendidos
 * entre 10 y 55 (incluidos), pares, y que no son ni el 16 ni múltiplos de 3.
 *
 * Seguro que al revisar detenidamente las posibilidades has descubierto algo nuevo.
 
 """
 
 #OPERADORES ARITMÉTICOS
 
print("Suma:", a + b)
print("Resta:", a - b)
print("Multiplicación:", a * b)
print("División:", a / b)
print("División entera:", a // b)
print("Módulo:", a % b)
print("Potencia:", a ** b)

# OPERADORES DE ASIGNACIÓN COMPUESTA
x = 10
x += 5
print("x += 5 →", x)
x *= 2
print("x *= 2 →", x)

# OPERADORES DE IDENTIDAD
lista1 = [1, 2, 3]
lista2 = lista1
lista3 = [1, 2, 3]
print("lista1 is lista2:", lista1 is lista2)
print("lista1 is lista3:", lista1 is lista3)
print("lista1 is not lista3:", lista1 is not lista3)

# OPERADORES DE PERTENENCIA
print("2 in lista1:", 2 in lista1)
print("4 not in lista1:", 4 not in lista1)

#EJEMPLOS DE OPERADORES LOGICOS 

edad_usuario=int(input("Introduce tu edad: "))
entrada=input("Has pagado la entrada? (si/no): ").lower()
if edad_usuario>=18 and entrada =="si":
    print("Bienvenido a la fiesta ")
else :
    print("Lo siento, no puedes entrar a la fiesta")

edad_cine=int(input("Introduce tu edad: "))
entrada=input("Tiene entrada VIP? (si/no): ").lower()
if edad_cine>=16 or entrada=="si":
    print("Puedes entrar al cine")
else:
    print("Lo siento, no puedes entrar al cine")
    
#OPERADORES DE COMPARACION
print("a == b:", a == b)
print("a != b:", a != b)
print("a > b:", a > b)
print("a < b:", a < b)
print("a >= b:", a >= b)
print("a <= b:", a <= b)

#ESTRUCTURAS DE CONTROL CONDICIONALES
edad=20
if edad>=18:
    print("Eres mayor de edad")
elif edad==17:
    print("Casi eres mayor de edad ")
else:
    print("Eres menor de edad")
    
print("Bucle while:")
i = 0
while i < 5:
    print(i)
    i += 1

print("Bucle for:")
for i in range(5):
    print(i)

# CONTROL DE FLUJO
print("Break:")
for i in range(10):
    if i == 5:
        break
    print(i)

print("Continue:")
for i in range(10):
    if i == 5:
        continue
    print(i)

print("Pass:")
for i in range(3):
    if i == 1:
        pass
    print(i)
# MANEJO DE EXCEPCIONES
try:
    numero = int(input("Introduce un número para dividir: "))
    resultado = 10 / numero
    print("Resultado:", resultado)
except ZeroDivisionError:
    print("No puedes dividir entre cero.")
except ValueError:
    print("Por favor, ingresa un número válido.")

# DIFICULTAD EXTRA: imprimir números del 10 al 55, pares, no 16 ni múltiplos de 3
print("Números del 10 al 55 que son pares, no son 16 y no son múltiplos de 3:")
for i in range(10, 56):
    if i % 2 == 0 and i != 16 and i % 3 != 0:
        print(i)